api/pkg/artifact/artifact.go

```diff
@@ -69,6 +80,37 @@ func (gac *GcsArtifactClient) ParseURL(gsURL string) (*URL, error) {
 	}, nil
 }
 
+type Service interface {
+	GetType() string
+	GetURLScheme() string
+	ParseURL(gsURL string) (*URL, error)
+	ReadArtifact(ctx context.Context, url string) ([]byte, error)
+	WriteArtifact(ctx context.Context, url string, content []byte) error
+	DeleteArtifact(ctx context.Context, url string) error
+}
+
+type GcsArtifactClient struct {
+	URLScheme
+	Type string
+	API  *storage.Client
+}
+
+func NewGcsArtifactClient() (Service, error) {
+	api, err := storage.NewClient(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("%s,failed initializing gcs for the artifact client", err.Error())
+	}
```

Comment:
I made a slight change here where the initialisation of the Google client is done by this function instead of relying on the caller to pass a Google client object. This simplifies the job of the component calling this function to initialise  a `GcsArtifactClient`.